src/fields/CoordinatesInput/DDCoordinatesInput.tsx

```diff
@@ -0,0 +1,116 @@
+import { debounce } from 'lodash.debounce'
+import { useCallback, useMemo, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import { isNumeric } from '../../utils/isNumeric'
+
+type DDCoordinatesInputProps = {
+  coordinates: [number, number]
+  onChange: (nextCoordinates: [number, number], coordinates: [number, number]) => void
+}
+export function DDCoordinatesInput({ coordinates, onChange }: DDCoordinatesInputProps) {
+  const latitudeInputRef = useRef<HTMLInputElement>()
+  const longitudeInputRef = useRef<HTMLInputElement>()
+
+  const [latitudeError, setLatitudeError] = useState('')
+  const [longitudeError, setLongitudeError] = useState('')
+
+  const defaultValue = useMemo(() => {
+    const [latitude, longitude] = coordinates
+
+    if (isNumeric(latitude) && isNumeric(longitude)) {
+      return {
+        latitude: Number(latitude),
+        longitude: Number(longitude)
+      }
+    }
+
+    return {
+      latitude: undefined,
+      longitude: undefined
+    }
+  }, [coordinates])
+
+  const handleChange = () => {
+    if (!latitudeInputRef.current || !longitudeInputRef.current) {
+      return
+    }
+
+    const latitudeAsString = latitudeInputRef.current.value
+    const longitudeAsString = longitudeInputRef.current.value
+
+    setLongitudeError('')
+    setLatitudeError('')
+
+    if (!isNumeric(latitudeAsString)) {
+      setLatitudeError('Champ Latitude incorrect')
+
+      return
+    }
+
+    if (!isNumeric(longitudeAsString)) {
+      setLongitudeError('Champ Longitude incorrect')
+
+      return
+    }
+
+    const latitude = Number(latitudeAsString)
+    const longitude = Number(longitudeAsString)
+
+    onChange([latitude, longitude], coordinates)
+  }
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedHandleChange = useCallback(debounce(handleChange, 500), [coordinates, onChange])
```

Comment:
I prefer we use https://github.com/xnimorz/use-debounce here because the implementation is more complex than "just" that.